### PR TITLE
test: replace minio with moto for testing AWS infra

### PR DESCRIPTION
**Pull request by vigith**
Signed-off-by: Vigith Maurice <vigith@gmail.com>

we will delete minio in a separate PR.

### CODE REVIEW

**Review by alexec — COMMENTED** [review]
can you delete the minio manifests too?

**Comment by alexec** [comment]
maybe we should rename the `*s3*` tests to be `*aws*` tests?

**Comment by vigith** [comment]
> Maybe we should rename the `*s3*` tests to be `*aws*` tests? That way we can test them all together.

should we create sub-dirs under aws for each aws resource?

**Comment by vigith** [comment]
I am not sure why `106-git-nodejs` is failing. 

```
2021/08/31 21:11:03 pipeline "106-git-nodejs" is Running "1 running" ["Running"]
2021/08/31 21:12:33 timeout waiting for condition SunkMessages
    examples_test.go:104: 
        	Error Trace:	examples_test.go:104
        	            				examples_test.go:51
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	Test/106-git-nodejs
```

**Comment by vigith** [comment]
> can you delete the minio manifests too?

done

File: test/s3-e2e/s3.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awscfg "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"strings"
+)
+
+func CreateBucket(bucket string) func() {
```
Comment (alexec) [inline test/s3-e2e/s3.go:14]: can you make sure you delete testapi/minio.go if we don't need it's endpoints anymore

Comment (alexec) [reply test/s3-e2e/s3.go:14]: or alternatively you might want to use that code directly

Comment (vigith) [reply test/s3-e2e/s3.go:14]: done

**Comment by alexec** [comment]
> should we create sub-dirs under aws for each aws resource?

I don't think we need to do that. The reason for having separate directories is so that we can avoid spinning up pods we don't need. E.g. starting MySQL when we are testing AWS.

**Pull request merged**